api/client/image/build.go

```diff
@@ -162,6 +162,7 @@ func runBuild(dockerCli *client.DockerCli, options buildOptions) error {
 		if err != nil && !os.IsNotExist(err) {
 			return err
 		}
+		defer f.Close()
```

Comment:
`ReadAll` closes the file, so this should not be necessary.

It might make sense to change `ReadAll` to take an `io.Reader` and then do the close here instead, though.